mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/impl/MongoDBDatastoreProvider.java

```diff
@@ -147,6 +154,19 @@ public void stop() {
 	}
 
 	public MongoDatabase getDatabase() {
+		if ( null == mongoDb ) {
+			synchronized ( this ) {
+				if ( null == mongoDb ) {
+					mongoDb = extractDatabase( mongo, config );
```

Comment:
Is this double-check necessary? Can't we use the SIngletone approach for lazy initialization? I'm referring to this article: http://blog.crazybob.org/2007/01/lazy-loading-singletons.html

Reply:
We can use the IODH if you're not worried about class loading through reflection as the same thread points out. Just tell me if you want and I'll do the change.

Reply:
I don't think it's going to be a problem in this context. We can revert it back later if I'm wrong